src/Altinn.Common/Altinn.EFormidlingClient/Altinn.EFormidlingClient/FormidlingClient.cs

```diff
@@ -0,0 +1,385 @@
+﻿using System;
+using System.IO;
+using System.Net;
+using System.Net.Http;
+using System.Net.Http.Headers;
+using System.Text;
+using System.Threading.Tasks;
+using Altinn.EFormidlingClient.Configuration;
+using Altinn.EFormidlingClient.Models;
+using Altinn.EFormidlingClient.Models.SBD;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+using Newtonsoft.Json.Serialization;
+
+namespace Altinn.EFormidlingClient
+{
+    /// <summary>
+    /// Represents an implementation of <see cref="IFormidlingClient"/> using a HttpClient.
+    /// </summary>
+    public class FormidlingClient : IFormidlingClient
+    {
+        private readonly HttpClient _client;
+        private readonly ILogger<FormidlingClient> _logger;
+        private readonly IOptions<EFormidlingClientSettings> _appSettings;
+
+        /// <summary>
+        /// Initializes a new instance of the IFormidlingClient class with the given HttpClient, lSettings and Logger.
+        /// </summary>
+        /// <param name="client">A HttpClient provided by a HttpClientFactory.</param>
+        /// <param name="appSettings">The settings configured for eFormidling package</param>
+        /// <param name="logger">Logging</param>
+        public FormidlingClient(HttpClient client, IOptions<EFormidlingClientSettings> appSettings, ILogger<FormidlingClient> logger = null)
+        {
+            _client = client ?? throw new ArgumentNullException("httpClient");
+            _appSettings = appSettings ?? throw new ArgumentNullException("appSettings");
+            _logger = logger ?? throw new ArgumentNullException("logger");
+
+            _client.DefaultRequestHeaders.Clear();
+            _client.DefaultRequestHeaders.Accept.Add(new MediaTypeWithQualityHeaderValue("application/json"));
+            _client.BaseAddress = new Uri(appSettings.Value.BaseUrl);
+        }
+
+        /// <inheritdoc/>
+        public async Task<bool> SendMessage(string id)
+        {
+            string responseBody;
+            if (string.IsNullOrEmpty(id))
+            {
+                throw new ArgumentNullException();
+            }
+           
+            try
+            {
+                HttpResponseMessage response = await _client.PostAsync($"messages/out/{id}", null);
+                responseBody = await response.Content.ReadAsStringAsync();
+
+                return true;
+            }
+            catch (HttpRequestException e)
+            {
+                _logger.LogError("Message :{0} ", e.Message);
```

Comment:
The exception should probably be sent to the app as well, so it is easier to handle. Consider creating a new class that can hold properties Successful (bool) and exception. Is it possible to make it generic enough to reuse for all the methods ?

Reply:
We could also consider generating a new ExceptionClass that is used for all exceptions returned from this client. Check out the implementation of PlatformHttpException in AppRef to see if a structure like that could work.